WordPress/Classes/ViewRelated/Voice/AudioRecorderVisualizerView.swift

```diff
@@ -0,0 +1,71 @@
+import SwiftUI
+import AVFoundation
+
+struct AudioRecorderVisualizerView: View {
+    let recorder: AVAudioRecorder
+    @StateObject private var viewModel = AudioRecorderVisualizerViewModel()
```

Comment:
nitpick;

To avoid an optional recorder within view model it could be passed through the initializer as well, and not on `onAppear`

```
    @StateObject private var viewModel: AudioRecorderVisualizerViewModel

    init(recorder: AVAudioRecorder) {
        _viewModel = StateObject(wrappedValue: AudioRecorderVisualizerViewModel(recorder: recorder))
    }
```

Reply:
It's a good technique, and I occasionally use it, but it has some potentially side effects due to how view identity works. It should be OK in this case, but I'll probably keep it as is for now. I'm surprised Apple put it in the official docs https://developer.apple.com/documentation/swiftui/stateobject#Initialize-state-objects-using-external-data.